index.js

```diff
@@ -14,14 +14,25 @@ var builder = require('./lib/builder');
 
 var path = require('path');
 
-/* Very basic, null-friendly, shallow clone for object attributes only */
-function clone(obj) {
-  obj = obj || {};
-  var copy = {};
-  for (var key in obj) {
-    copy[key] = obj[key];
-  }
-  return copy;
+function processOpts(opts_, outFile) {
+  var keys = [
+    'config',
+    'lowResSourceMaps',
+    'minify',
+    'normalize',
+    'outFile',
+    'runtime',
+    'sourceMaps',
+    'sourceMapContents'
+  ];
+  var opts = {};
+  keys.forEach(function(key) {
+    opts[key] = opts_ && opts_[key];
+  });
```

Comment:
Almost added it as `extend(defaults, normalize(opts), overrides)`, but then stopped because really config should also be normalized, and that triggered my subscription to the religion that configuration should never be nested, so then I went back to "smallest thing that meets this nitpickers explicit points" :japanese_ogre: 

Sure we could add this prepend form as a "stopgap" ;)